### PR TITLE
Provide generic tts integration

### PR DESCRIPTION
**Pull request by raviks789**
ref #629

### CODE REVIEW

**Comment by nilmerg** [comment]
Planned for v1.1. Once working on this version, I'd like to discuss whether it makes sense to re-use some existing hooks or whether to introduce entirely new hooks. (e.g. `CommentDetailExtensionHook`) Why? Because I'd like to get rid of `TicketHook` in Icinga Web 2 and have its implementation provided by generictts. (What the missing support for it in Icinga DB Web showed, is that it's the only module making use of the hook, so why the hell not..)

**Comment by lippserd** [comment]
A new hook like `CommentDetailExtensionHook` sounds good to me as well!

File: library/Icingadb/Common/TicketLinks.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
```
Comment (nilmerg) [inline library/Icingadb/Common/TicketLinks.php:3]: It's 2022 ;)

File: library/Icingadb/Common/TicketLinks.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    /**
+     * Get whether list items should render host and service links
+     *
+     * @return string
+     */
+    public function createTicketLinks($text): string
+    {
+        $tickets = Icinga::app()->getViewRenderer()->view->tickets;
```
Comment (nilmerg) [inline library/Icingadb/Common/TicketLinks.php:45]: I don't quite see an advantage in this. Just retrieve the hook here directly by calling `Hook::first()`.

File: library/Icingadb/Common/TicketLinks.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        if ($this->getTicketLinkEnabled()) {
+            /** @var \Icinga\Application\Hook\TicketHook $tickets */
+            return isset($tickets) ? $tickets->createLinks($text) : $text;
```
Comment (nilmerg) [inline library/Icingadb/Common/TicketLinks.php:49]: Why not move the `isset()` into the `if`?

File: library/Icingadb/Web/Controller.php [hunk 4]
```diff
@@ -562,5 +563,9 @@ protected function moduleInit()
         Icinga::app()->getFrontController()
             ->getPlugin('Zend_Controller_Plugin_ErrorHandler')
             ->setErrorHandlerModule('icingadb');
+
+        if (Hook::has('ticket')) {
+            $this->view->tickets = Hook::first('ticket');
+        }
```
Comment (nilmerg) [inline library/Icingadb/Web/Controller.php:569]: Please remove this.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/Web/Controller.php [hunk 5]
```diff
@@ -8,6 +8,7 @@
 use Generator;
 use GuzzleHttp\Psr7\ServerRequest;
 use Icinga\Application\Config;
+use Icinga\Application\Hook;
```
Comment (nilmerg) [inline library/Icingadb/Web/Controller.php:11]: There's still this now unused use statement

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]
Thanks!

**Pull request merged**